From 5afbc52c470940f0935b352b283bf4a5b61db146 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Thu, 13 Sep 2018 23:19:07 +0200
Subject: [PATCH] Don't use the internal glibc header sys/cdefs.h

This patch was created using sed(1) and ag(1) with the following script:

	for file in $(ag -l __BEGIN_DECLS); do
		sed -i "${file}" \
			-e 's/__BEGIN_DECLS/#ifdef __cplusplus\nextern "C" {\n#endif/g' \
			-e 's/__END_DECLS/#ifdef __cplusplus\n}\n#endif/g' \
			-e '/#include <sys\/cdefs.h>/d'
	done

https://wiki.musl-libc.org/faq.html#Q:-When-compiling-something-against-musl,-I-get-error-messages-about-<code>sys/cdefs.h</code>
---
 libfscrypt/include/fscrypt/fscrypt.h  | 9 ++++++---
 tests/lib/testUtil/include/testUtil.h | 8 ++++++--
 2 files changed, 12 insertions(+), 5 deletions(-)

diff --git a/libfscrypt/include/fscrypt/fscrypt.h b/libfscrypt/include/fscrypt/fscrypt.h
index 13358bb7..b77999a4 100644
--- a/libfscrypt/include/fscrypt/fscrypt.h
+++ b/libfscrypt/include/fscrypt/fscrypt.h
@@ -17,11 +17,12 @@
 #ifndef _FSCRYPT_H_
 #define _FSCRYPT_H_
 
-#include <sys/cdefs.h>
 #include <stdbool.h>
 #include <cutils/multiuser.h>
 
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 bool fscrypt_is_native();
 
@@ -34,6 +35,8 @@ static const char* fscrypt_key_ref = "/unencrypted/ref";
 static const char* fscrypt_key_per_boot_ref = "/unencrypted/per_boot_ref";
 static const char* fscrypt_key_mode = "/unencrypted/mode";
 
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
 #endif // _FSCRYPT_H_
diff --git a/tests/lib/testUtil/include/testUtil.h b/tests/lib/testUtil/include/testUtil.h
index 3b75914d..bdef437d 100644
--- a/tests/lib/testUtil/include/testUtil.h
+++ b/tests/lib/testUtil/include/testUtil.h
@@ -22,7 +22,9 @@
 #include <stdio.h>
 #include <sys/time.h>
 
-__BEGIN_DECLS
+#ifdef __cplusplus
+extern "C" {
+#endif
 
 // Time Utilities
 struct timespec double2ts(double amt);
@@ -64,6 +66,8 @@ uint64_t testXDumpGetOffset(void);
 // Command Execution
 void testExecCmd(const char *cmd);
 
-__END_DECLS
+#ifdef __cplusplus
+}
+#endif
 
 #endif
